include/alpaka/vec/Vec.hpp

```diff
@@ -186,8 +186,8 @@ namespace alpaka
                         return false;
                     }
                 }
+                return true;
```

Comment:
What is the reason for this change? I believe this will cause `unreachable code` warnings with some known buggy compilers.

Reply:
Actually it should be the opposite. The old code
```c++
            if constexpr(TDim::value == 0)
            {
                return true;
            }
            else
            {
                ...
            }
            return true;
```

has an unreachable `return true` at the end if `TDim::value == 0`.
Moving it into the `else` clause should fix it:
```c++
            if constexpr(TDim::value == 0)
            {
                return true;
            }
            else
            {
                ...
                return true;
            }
```

Reply:
Ah, I phrased that wrong. The old code would indeed cause an unreachable code warning (and rightly so). The new code will cause the buggy compilers to complain though. `nvcc` (before CUDA 11.5) will believe that there is now a missing return statement at the end of the function because it is unable to correctly identifiy `if constexpr` code paths. That is why we recently introduced `ALPAKA_UNREACHABLE` to address this:

```suggestion
                return true; // no problem for conforming compilers
            }
            ALPAKA_UNREACHABLE(bool{}); // silence nvcc < 11.5 and other buggy compilers
        }
```

Reply:
Ah, I see.